src/syncemailclient.cpp

```diff
@@ -86,8 +80,15 @@ bool SyncEmailClient::uninit()
 
 bool SyncEmailClient::startSync()
 {
-    qDebug() << Q_FUNC_INFO << "Starting scheduled sync for email account: " << m_accountId.toULongLong();
-    m_emailAgent->syncAccounts(QMailAccountIdList() << m_accountId);
+    if (m_emailAgent->ipcConnected()) {
+        triggerSync();
+    } else {
+        qWarning() << Q_FUNC_INFO << "IPC not connect yet, waiting....";
+        connect(m_emailAgent, SIGNAL(ipcConnectionEstablished()), this, SLOT(ipcConnected()));
+        // Since the process running this plugin can cause a wakelock we dont want to wait long time for IPC connection
+        connect(&m_ipcTimeout, SIGNAL(timeout()), this, SLOT(ipcTimeout()));
```

Comment:
So after it syncs, the process exits?  This could only be a problem if this process stayed alive while the messageserver restarted.